File: Algo/Java/CountAndSay.java
/* https://leetcode.com/problems/count-and-say/ */

public class CountAndSay
{
	public String countAndSay(int n) 
	{
		String num = 1+"";
        for(int i=1;i<n;i++)
        {
        	num = helper(num);
        }
        return num;
    }

    public String helper(String num)
    {
    	String s="";
    	int count = 1;
    	char prevChar = num.charAt(0);
    	for(int i=1;i<num.length();i++)
    	{
    		if(num.charAt(i)==prevChar)
    		{
    			++count;
    		}
    		else
    		{
    			s += count+""+prevChar;
    			prevChar = num.charAt(i);
    			count = 1;
    		}
    	}
    	s+=count+""+prevChar;
    	return s;
    }

    public static void main(String args[])
    {
    	CountAndSay cs = new CountAndSay();
    	System.out.println(cs.countAndSay(1));
    }
}

File: Algo/python/CountAndSay.py
''' https://leetcode.com/problems/count-and-say/ '''

class Solution(object):
    def getNext(self, s):
        p = s[0]
        c = 0
        res = ''
        for ch in s:
            if ch == p:
                c += 1
            else:
                res += str(c) + p
                p = ch
                c = 1
        res += str(c) + p
        return res
                
    def countAndSay(self, n):
        """
        :type n: int
        :rtype: str
        """
        if n == 1:
            return "1"
        s = '1'
        while n > 1:
            s = self.getNext(s)
            n -= 1
            
        return s
